bundles/binding/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/LcnBinding.java

```diff
@@ -0,0 +1,187 @@
+	
+	/** Holds all currently existing connections. */
+	private ConnectionManager connections = new ConnectionManager(this);	
+	
+	/** Semaphore to wait for new notifications to process in refresh thread. */
+	private Semaphore notificationsSem = new Semaphore(0);
+	
+	/**
+	 * List of notifications to process in the refresh thread.
+	 * List must be synchronized.
+	 */
+	private final LinkedList<LcnBindingNotification> notifications = new LinkedList<LcnBindingNotification>(); 
+	
+	/**
+	 * Called whenever the configurations are updated.
+	 * 
+	 * @param config the updated configurations
+	 */
+	@Override
+	public void updated(Dictionary<String, ?> config) throws ConfigurationException {
+		LcnLogger.info("Loading LCN configuration...");
+		// Reset to default values
+		this.connectionSettings.clear();
+		// Load connection settings
+		int counter = 0;
+		ConnectionSettings sets;
+		while ((sets = ConnectionSettings.tryParse(config, counter)) != null) {
+			this.connectionSettings.put(sets.getId().toUpperCase(), sets);
+			++counter;
+		}
+		LcnLogger.info("LCN configuration loaded.");
+		// Finished
+		this.setProperlyConfigured(true);
+	}
+	
+	/** {@inheritDoc} */
+	@Override
+	protected void internalReceiveCommand(String itemName, Command command) {
+		final String n = itemName;
+		final Command c = command;
+		this.runOnRefreshThreadAsync(new LcnBindingNotification() {
+			public void execute() {
+				for (LcnBindingProvider provider : LcnBinding.this.providers) {
+					LcnBindingItem itemConfig = provider.getLcnItemConfig(n);
+					if (itemConfig != null)
+						itemConfig.send(LcnBinding.this.connections, c);
+				}
+			}
+		});
+	}
+	
+	/**
+	 * Tells the connections the binding is activated.
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void activate() {
+		this.connections.activate();
+	}
+
+	/**
+	 * Tells the connections the binding is deactivated.
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void deactivate() {
+		this.connections.deactivate();
+	}
+	
+	/**
+	 * Processes notifications to be run on the refresh thread.
+	 * {@inheritDoc}
+	 */
+	@Override
+	protected void execute() {
+		// Update connections
+		this.connections.update();
+		this.connections.flush();
+		// Process notifications
+		try {
+			long startTime = System.nanoTime();
+			int timeoutMSec = REFRESH_INTERVAL_MSEC;
+			while (timeoutMSec >= 0 && this.notificationsSem.tryAcquire(timeoutMSec, TimeUnit.MILLISECONDS)) {
+				LcnBindingNotification n;
+				synchronized (this.notifications) {
+					n = this.notifications.pollFirst();
+				}
+				n.execute();
+				this.connections.flush();
+				// Next
+				timeoutMSec = REFRESH_INTERVAL_MSEC - (int)((System.nanoTime() - startTime) / 1000000L);
+			}
+		} catch (InterruptedException ex) { }
+	}
+	
+	/** {@inheritDoc} */
+	@Override
+	protected long getRefreshInterval() {
+		return 0;  // Will cause delays between execute() calls. We don't need that.
```

Comment:
you have deactivated waiting here and implemented your own "waiting" in execute() (while timeoutMSec >=0). since the the Thread continuously hits while this seems a "busy waiting" implementation. Why not use the "normal" refresh interval and remove your implementation from execute()?